torch/onnx/_internal/exporter.py

```diff
@@ -89,6 +89,44 @@ class OnnxRegistry:
     fixed opset version. It supports registering custom onnx-script functions and for
     dispatcher to dispatch calls to the appropriate function.
 
+    Example::
+
+        >>> # xdoctest: +SKIP("RuntimeError: Decorator script does not work on dynamically compiled function foo_onnx.")
+        >>> import torch
+        >>> import torch.onnx
+        >>> from torch._custom_op import impl as custom_op
+        >>> import onnxscript
+        >>> @custom_op.custom_op("mylibrary::foo_op", "foo_op(Tensor x) -> Tensor")
+        >>> def foo_op(x):
```

Comment:
We always encourage type hinting, but for custom ops API we have removed the type hint from the onnx function and added a string to `custom_op` with the expected typing. 

Is there a way to keep relying on type hinting? This is the most pythonic and intuitive way for users
What happens if the user do type hinting instead? Is there an error message to let them know where is the right place to declare type hint?
What happens if the user do type hinting and also use the type string on `custom_op`? Can we detect this scenario

Reply:
> Might as well revert the workaround now that the example is skipped anyway due to `script`.
> 
Done

> On second thought, maybe we can demonstrate on aten op directly. The torch custom op took half the real estate here and is not our api.
> 
Use aten op to demo doesn't come in handy, because aten op is the target we will always try to record into torchlib, which means I will have to pop the example op out in the example, and that would be confused. I think we should stick to this example, and I put "Steps" to explain more what is goingo on between them. I think would help more. The very similar example is also written in test_fx_to_onnx.py to guard it.

> The `script` issue is problematic. It is expected for users to copy the example and run by themselves, If they run in interpreter directly it would fail.

I will see if I can understand the issue and fix. https://github.com/microsoft/onnxscript/issues/82

Reply:
The docstring is using type hinting, but both unit test files are still using the string argument at `custom_op`.

Reply:
The test files should remain the way they are until `from __future__ import annotations` is supported in torch custom_op.

Reply:
ok, so the test must remain in the old way for now, but will the user reading the docstring be able to use the type hint style? Otherwise, we are publishing a docstring with non-functional code

Reply:
Yes, they will. In fact, they should provide either type int or manual schema to create a custom op in torch. The best I can do is to revert back the test in `test_fx_pass.py` to type hint, and delete from __future__ import annotations, if we want to know if the custom op registration is changed in the future.